### ISSUE TITLE
[Feat] t.rast.algebra: add time functions (start_doy, start_year, ...) that are currently unsupported in the spatial part of the expression

### ISSUE THREAD

**Issue by ninsbl** [OSGeo/grass#891]
**Describe the bug**
According to function documentation, [time functions](https://github.com/OSGeo/grass/blob/master/lib/python/temporal/temporal_algebra.py#L506) that return a float or integer (like e.g. start_doy, start_year) should be usable as constants in the spatial part of the expression. However, that does not seem to work, currently.
If e.g. one needs the day of the year as a numerical constant in t.rast.algebra (for being able to use it along with temporal selection and temporal windows) one has to run t.rast.algebra first and then t.rast.mapcalc. That causes unnecessary processing overhead.
The time delta (td) time function works partly, maybe because of this(?): https://github.com/OSGeo/grass/blob/master/lib/python/temporal/temporal_raster_base_algebra.py#L322
t.rast.algebra is a quite involved module and I do not fully get my head around how library functions are interrelated, so any help is highly appreciated....

**To Reproduce**

Using artificial maps from the testsuite, this
```
t.rast.algebra -d basename=r expression="R=start_doy(A)"
t.rast.algebra -d basename=r expression="R=A+start_doy(A)"
t.rast.algebra -d basename=r expression="R=if(A>12,start_doy(A),0)"
t.rast.algebra -d basename=r expression="R=td(A)"
t.rast.algebra -d basename=r expression="R=if(A>12,td(A),0)"
```
does not work (syntax error at e.g. token START_DOY), while this:

```
t.rast.algebra -d basename=r expression="R=A+td(A)"
t.rast.algebra -d basename=r expression="R=if(A>12,1,0)*td(A)"
t.rast.algebra -d basename=r expression="R=if(A>12,A+td(A),0)"
t.rast.algebra -d basename=r expression="R=if(A>12 && A[1]>12,A+td(A),0)"
```
does work and returns reasonable mapcalculator exressions.

**Expected behavior**
[Time functions](https://github.com/OSGeo/grass/blob/master/lib/python/temporal/temporal_algebra.py#L506) that return a float or integer (like e.g. start_doy, start_year) should be evaluated to numerical constants and included in the spatial part of the expression.

I would like to be able to do something like:
```
# Return day of the year when temperature (A) exceeds a certain value in three consecutive days
t.rast.algebra basename=r expression="R=if(start_month(A)<9, if(A>50 && A[1]>50 && A[2]>50, start_doy(A), 0))"
```
Or alternatively:
```
# Return day of the year when temperature (A) exceeds a certain value in three consecutive days
t.rast.algebra basename=r expression="R=if(start_month(A)<9, if(A>50 && A[1]>50 && A[2]>50, 1, 0)*start_doy(A))"
```

**System description (please complete the following information):**

- Operating System: Ubuntu Linux
- GRASS GIS version: 7.9

**Additional context**

**Comment by veroandreo** [comment]
I can confirm this error. Usually I have used these temporal variables in `t.rast.mapcalc` without passing the STDS within the parenthesis, but indeed the manual of `t.rast.algebra` says that the STDS should be passed there. 

Maybe @huhabla or @m4sth0 can explain if this is indeed a bug or the intended/expected behaviour? In any case, the use you propose would be really cool

**Comment by ninsbl** [comment]
OK, I think I have a solution. Will create a PR soon.

**Comment by ninsbl** [comment]
Since this actually never worked it is probably a feature rather than a bug?

**Comment by veroandreo** [comment]
Well, in the facts, yes... the manual of `t.rast.algebra` indicates the way to use those temporal variables for a while though... so, I'm not sure if we can consider it a new feature. I guess the module is so complex that it didn't get much testing.

**Comment by veroandreo** [comment]
I correct myself here, what has been implemented in #905 seems indeed to be a new feature, but IIUC, it is not the same that was reported originally here. This `t.rast.algebra -d basename=r expression="R=start_doy(A)"` still does not work for me.

**Comment by ninsbl** [comment]
Fixed in https://github.com/OSGeo/grass/pull/905
